test/integration/extension_discovery_integration_test.cc

```diff
@@ -775,5 +855,79 @@ TEST_P(ExtensionDiscoveryIntegrationTest, ReloadBoth) {
   codec_client_->close();
 }
 
+// ECDS config dump test with one listener ECDS filter and one HTTP ECDS filter.
+TEST_P(ExtensionDiscoveryIntegrationTest, ConfigDumpWithTwoSubscriptionTypes) {
+  DISABLE_IF_ADMIN_DISABLED; // Uses admin interface.
+  two_ecds_filters_ = true;
+  on_server_init_function_ = [&]() { waitXdsStream(); };
+  // HTTP ECDS filter
+  addDynamicFilter("foo", false);
+  // Listener ECDS filter
+  addDynamicListenerFilter("bar");
+  initialize();
+
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 1);
+  EXPECT_EQ(test_server_->server().initManager().state(), Init::Manager::State::Initializing);
+  registerTestServerPorts({"http"});
+
+  // Send configuration update for HTTP ECDS filter.
+  sendXdsResponse("foo", "1", denyPrivateConfig());
+  // Send configuration update for listener ECDS filter.
+  sendListenerFilterEcdsResponse("bar", "2", 7);
```

Comment:
Are they both sending xDS responses, but one on stream1 and the other on stream2?
If so, I suggest rewriting the method names/generalizing them, so it will be more clear.

Reply:
done